### ISSUE TITLE
Why is the result of my training llama-13b-hf using scripts/run_raft_align.sh very poor?

### ISSUE THREAD

**Issue by haley05-cloud** [OptimalScale/LMFlow#440]
Why is the result of my training llama-13b-hf using scripts/run_raft_align.sh very poor? Basically, none of the questions were answered correctly. Below is my script. Is there something wrong of my script?

#!/bin/bash

Please run this script under project directory.
deepspeed_args="--master_port=11000" # Default argument
if [ $# -ge 1 ]; then
deepspeed_args="$1"
fi

exp_id=raft_align
project_dir=$(cd "$(dirname $0)"/..; pwd)
output_dir=${project_dir}/output_llama_13b/${exp_id}
log_dir=${project_dir}/log/${exp_id}

mkdir -p ${output_dir} ${log_dir}

echo "exp_id=${exp_id}, project_dir=${project_dir}, output_dir=${output_dir}, log_dir=${log_dir}"

export PYTHONPATH=.
deepspeed ${deepspeed_args}
examples/raft_align.py
--model_name_or_path /home/user01/.cache1/llama-13b-hf
--num_raft_iteration 2
--learning_rate 2e-4
--lr_scheduler_type "constant"
--bf16
--deepspeed configs/ds_config_zero2.json
--dataset_path ${project_dir}/data/imdb/train
--output_reward_path ${project_dir}/tmp/raft_aligner/reward.txt
--output_dir ${output_dir} --overwrite_output_dir
--run_name ${exp_id}
--num_train_epochs 4
--per_device_train_batch_size 1
--per_device_eval_batch_size 1
--validation_split_percentage 0
--logging_steps 1
--do_train
--ddp_timeout 72000
--save_steps 35
--dataloader_num_workers 1
--preprocessing_num_workers 12
--inference_batch_size_per_device 1
| tee ${log_dir}/raft_align.log
2> ${log_dir}/raft_align.err

example
![image](https://github.com/OptimalScale/LMFlow/assets/85667300/8f76fb4b-8a22-4317-90b4-2f5a79bde31f)

**Comment by WeiXiongUST** [comment]
Hi! Thanks for your feedback.

It seems that you are trying to use the model as a chatbot. However, you use the dataset 
--dataset_path ${project_dir}/data/imdb/train
which will truncate the first several words as the prompt and ask the model to complete the movie review. Therefore, the training setting and your target seem not to match. In particular, the llama model does not perform well in answering the chat according to the human-assistant prompt if you do not SFT it first. Meanwhile, the learning rate 2e-4 is too large for the LLAMA model according to my personal experience (I will use 2e-5 typically). 

Actually, we are considering the imdb movie review completion task rather limited and plan to use the HH-RLHF dataset as a better example, which is more suitable for your purpose. For now, you may have a look at our updated version https://arxiv.org/pdf/2304.06767.pdf to get the idea of the HH-RLHF experiment. We will provide a step-by-step document this week to help the users to get started with and recover the results of RAFT on this experiment.

**Comment by WeiXiongUST** [comment]
Hi! We have updated the code accordingly and you can find the tutorial here https://optimalscale.github.io/LMFlow/examples/raft.html

We upload a reward model to the huggingface (based on GPT-Neo-2.7B and with ~68% test accuracy) and you may directly try out RAFT now (i.e., the last step of the tutorial) and then you may try to improve the model by involving SFT and a better reward model trained by yourself.

If you have any further questions or feedbacks, feel free to let us know!

**Comment by shizhediao** [comment]
This issue has been marked as stale because it has not had recent activity. If you think this still needs to be addressed please feel free to reopen this issue. Thanks
